examples/qml_features/rust/src/containers.rs

```diff
@@ -26,16 +33,34 @@ mod ffi {
         pub fn insert(mut self: Pin<&mut Self>, value: i32) {
             unsafe {
                 self.as_mut().rust_mut().set.insert(value);
+                self.as_mut()
+                    .rust_mut()
+                    .hash
+                    .insert(QString::from(&value.to_string()), QVariant::from(value * 2));
             }
 
+            let hash_items = self
+                .as_ref()
+                .hash()
+                .iter()
+                .map(|(key, value)| {
+                    let value = match value.value() {
+                        QVariantValue::I32(value) => value,
+                        _others => 0,
+                    };
+                    format!("{} => {}", key, value)
+                })
+                .collect::<Vec<String>>()
+                .join(", ");
+            self.as_mut().set_string_hash(QString::from(&hash_items));
```

Comment:
It's quite confusing that a single "Insert" button works on both the QSet and the QHash. How about making two new TextFields and adding another "Insert" button in the QML just for the QHash?

Reply:
I was thinking originally of having a single insert method that adds an entry into all container types then you can see how some end up with unique entries and some get duplicates etc. But maybe it's too confusing.

I'll see if i can think of a better example :thinking: As we are going to have QHash, QMap, QSet, QList, QVector.

Reply:
I've changed it so we have one row per container type, then an insert button next to them. Then the spinbox at the top is used for the value and there is a reset at the top.

A deeper overhaul could be later :-)